src/main/java/org/deepsymmetry/beatlink/MediaDetails.java

```diff
@@ -139,10 +139,16 @@ private int getUTF16StringLength(byte[] source, int offset, int maxLength) {
         this.slotReference = slotReference;
         this.mediaType = mediaType;
         this.name = name;
-        this.creationDate = Long.toString(db.sourceFile.lastModified());
-        this.trackCount = db.trackIndex.size();
+        if (db != null && db.sourceFile != null) {
```

Comment:
This will not be necessary once the `CdjStatus` packets are correctly reporting which devices have USBs mounted.

Reply:
Should I do that work in this PR or create another one? This one is already rather large.

Reply:
To the extent that this is a step in the wrong direction, I don’t want to merge it. It would have been better to submit multiple small PRs that we could have refined individually than all of this at once. But I don’t know if you would rather go that route now, separating out the non-problematic parts, or just fix it to merge as one?

Reply:
Ah! Since you rolled back the above change, you can roll this back too, and merge this first, and do the slots/devices change as a later PR.

Reply:
Sadly rolling this back will end up with NPEs on startup of BLT before weve loaded in any archives.

Reply:
Oh, fair enough. Adjusting and testing now.

Reply:
Sadly did not work, when I revert this the NPE comes back 

```quote
2024-May-07 23:56:44.352 18:3E:EF:DA:5B:CA WARN [org.deepsymmetry.beatlink.VirtualRekordbox:848] - Problem processing device update packet
                                  java.lang.Thread.run            Thread.java: 829
      org.deepsymmetry.beatlink.VirtualRekordbox$1.run  VirtualRekordbox.java: 842
org.deepsymmetry.beatlink.VirtualRekordbox.buildUpdate  VirtualRekordbox.java: 347
         org.deepsymmetry.beatlink.MediaDetails.<init>      MediaDetails.java: 142
java.lang.NullPointerException: 

2024-May-07 23:56:44.408 18:3E:EF:DA:5B:CA WARN [org.deepsymmetry.beatlink.VirtualRekordbox:848] - Problem processing device update packet
                                  java.lang.Thread.run            Thread.java: 829
      org.deepsymmetry.beatlink.VirtualRekordbox$1.run  VirtualRekordbox.java: 842
org.deepsymmetry.beatlink.VirtualRekordbox.buildUpdate  VirtualRekordbox.java: 347
         org.deepsymmetry.beatlink.MediaDetails.<init>      MediaDetails.java: 142
java.lang.NullPointerException: 

2024-May-07 23:56:44.544 18:3E:EF:DA:5B:CA WARN [org.deepsymmetry.beatlink.VirtualRekordbox:848] - Problem processing device update packet
                                  java.lang.Thread.run            Thread.java: 829
      org.deepsymmetry.beatlink.VirtualRekordbox$1.run  VirtualRekordbox.java: 842
org.deepsymmetry.beatlink.VirtualRekordbox.buildUpdate  VirtualRekordbox.java: 347
         org.deepsymmetry.beatlink.MediaDetails.<init>      MediaDetails.java: 142
java.lang.NullPointerException: 

2024-May-07 23:56:44.546 18:3E:EF:DA:5B:CA WARN [org.deepsymmetry.beatlink.VirtualRekordbox:848] - Problem processing device update packet
                                  java.lang.Thread.run            Thread.java: 829
      org.deepsymmetry.beatlink.VirtualRekordbox$1.run  VirtualRekordbox.java: 842
org.deepsymmetry.beatlink.VirtualRekordbox.buildUpdate  VirtualRekordbox.java: 347
         org.deepsymmetry.beatlink.MediaDetails.<init>      MediaDetails.java: 142
java.lang.NullPointerException: 
```

Reply:
The issue happens when there are no archives loaded at all. Once an archive is loaded the errors stop.

Reply:
FindDatabase returns null when there is no loaded Databases. We can throw instead, but that is less efficient with system resources and not technically an error.

Reply:
Happy to do it though if you prefer :).

Reply:
I don’t understand the question. It returns `null` to tell you there is no database mounted in the place you asked about which is correct information, and I agree it is not an error.

Reply:
This is no longer applicable. Resolving.